pkgs/stdenv/native/default.nix

```diff
@@ -1,4 +1,9 @@
-{ system, allPackages ? import ../../.., config }:
+{ _defaults    ? import ../debug.nix
+, argsResolved ? _defaults.extend (_: _: args)
+, allPackages  ? argsResolved.allPackages
```

Comment:
I honestly think the code is better without these debug code.
This debug code makes it much more harder to understand what is going on with the argument.

I think if you are an stdenv developer adding a few `builtins.trace` function call is probably the easiest.
As well as making a wrapper file to invoke the right stage of the bootstrap you are looking for.